index.js

```diff
@@ -10,11 +10,28 @@ var onExit = require('signal-exit')
 var path = require('path')
 var activeFiles = {}
 
+// if we run inside of a worker_thread, `process.pid` is not unique
+/* istanbul ignore next */
+var threadId = (function getId () {
+  try {
+    var workerThreads = require('worker_threads')
+
+    if (workerThreads.isMainThread) {
+      return 0
+    }
```

Comment:
https://github.com/nodejs/node/blob/8c9aaacb333a0223243eda9b3551dbf26fdb260a/lib/internal/worker.js#L36

Generally, if you can require `worker_threads`, then just returning `workerThreads.threadId` should be okay :)

Reply:
ah, awesome. simplifies things